pandas/tests/test_format.py

```diff
@@ -2952,6 +2952,16 @@ def test_to_csv_date_format(self):
         self.assertEqual(df_day.to_csv(), expected_default_day)
         self.assertEqual(df_day.to_csv(date_format='%Y-%m-%d'), expected_default_day)
 
+        # GH 7791
```

Comment:
Yeah, ok.

Hum, `float_format` does NOT work even for a simple, non-multi index (same as `decimal`, so). 

```
In [58]: pd.DataFrame({'a': [0.1,1.1], 'b': [2, 3]}).set_index('a').to_csv(float_format='%.2f')
Out[58]: 'a,b\n0.1,2\n1.1,3\n'
```

That's strange, because indeed the arguments that are passed are indeed `na_rep` `float_format`, `date_format`, `quoting`